tests/explainable/test_handlers_explanation.py

```diff
@@ -314,3 +316,236 @@ def test_SplitSubsets_two_layers():
         ).item()
 
     assert obs_bill_hits == 0.0 and int_bill_hits == 0.0 and int_bottle_shatters == 0.0
+
+
+def test_edge_eq_neq():
+    def model_independent():
+        X = pyro.sample("X", dist.Bernoulli(0.5))
+        Y = pyro.sample("Y", dist.Bernoulli(0.5))
+        return {"X": X, "Y": Y}
+
+    def model_connected():
+        X = pyro.sample("X", dist.Bernoulli(0.5))
+        Y = pyro.sample("Y", dist.Bernoulli(X))
+        return {"X": X, "Y": Y}
+
+    with ExtractSupports() as supports_independent:
+        model_independent()
+
+    with ExtractSupports() as supports_connected:
+        model_connected()
+
+    with MultiWorldCounterfactual():
+        with SearchForExplanation(
+            supports=supports_independent.supports,
+            antecedents={"X": torch.tensor(1.0)},
+            consequents={"Y": torch.tensor(1.0)},
+            witnesses={},
+            alternatives={"X": torch.tensor(0.0)},
+            antecedent_bias=-0.5,
+            consequent_scale=0,
+        ):
+            with pyro.plate("sample", size=3):
+                with pyro.poutine.trace() as trace_independent:
+                    model_independent()
+
+    with MultiWorldCounterfactual():
+        with SearchForExplanation(
+            supports=supports_connected.supports,
+            antecedents={"X": torch.tensor(1.0)},
+            consequents={"Y": torch.tensor(1.0)},
+            witnesses={},
+            alternatives={"X": torch.tensor(0.0)},
+            antecedent_bias=-0.5,
+            consequent_scale=0,
+        ):
+            with pyro.plate("sample", size=3):
+                with pyro.poutine.trace() as trace_connected:
+                    model_connected()
+
+    with MultiWorldCounterfactual():
+        with SearchForExplanation(
+            supports=supports_connected.supports,
+            antecedents={"Y": torch.tensor(1.0)},
+            consequents={"X": torch.tensor(1.0)},
+            witnesses={},
+            alternatives={"Y": torch.tensor(0.0)},
+            antecedent_bias=-0.5,
+            consequent_scale=0,
+        ):
+            with pyro.plate("sample", size=3):
+                with pyro.poutine.trace() as trace_reverse:
+                    model_connected()
+
+    trace_connected.trace.compute_log_prob
+    trace_independent.trace.compute_log_prob
+    trace_reverse.trace.compute_log_prob
+
+    Y_values_ind = trace_independent.trace.nodes["Y"]["value"]
+
+    log_probs_ind = trace_independent.trace.nodes["__cause____consequent_Y"][
+        "fn"
+    ].log_factor
+    if torch.any(Y_values_ind == 1.0):
+        assert log_probs_ind[1, 0, 0, 0, :].sum().exp() == 0.0
```

Comment:
All of these indexing operations should be replaced with equivalent calls to `gather` instead of hardcoding batch shapes, which are implementation details of `MultiWorldCounterfactual` and may change.